File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => [0,2,5,3,7,8,4];

    public int Today() => birdsPerDay[6];

    public void IncrementTodaysCount() => birdsPerDay[6] = Today() + 1;

    public bool HasDayWithoutBirds() => birdsPerDay.Contains(0);

    public int CountForFirstDays(int numberOfDays)
    {
        int count = 0;
        for (int i = 0; i < numberOfDays; i++)
        {
            count += birdsPerDay[i];
        }
        return count;
    }

    public int BusyDays()
    {
        int busyDay = 0;
        foreach(int birdCount in birdsPerDay)
        {
            if(birdCount >= 5) busyDay += 1;
        }
        return busyDay;
    }
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public static int[] getLastWeek() {
        return new int[] {0,2,5,3,7,8,4};
    }

    public int getToday() {
        int lenDays = birdsPerDay.length;
        return birdsPerDay[lenDays - 1];
    }

    public void incrementTodaysCount() {
        int lenDays = birdsPerDay.length;
        birdsPerDay[lenDays - 1] += 1; 
    }

    public boolean hasDayWithoutBirds() {

        int flag = 0;
        
        for (int count : birdsPerDay){
            
            if (count == 0){
                flag += 1;
            }
        }
        return flag > 0;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum = 0;

        int days = numberOfDays;

        if (numberOfDays > 7){
            days = 7;
        }
        
        for (int i = 0; i < days; i++){
            sum += birdsPerDay[i];
        }
        
        return sum;
    }

    public int getBusyDays() {
        
        int busyDays = 0;

        for (int count : birdsPerDay){
            if (count >= 5) {
                busyDays += 1;
            }
        }
        
        return busyDays;
    }
}
